### DISCUSSION TITLE
DYP-A02YY ultrasonic sensor & WeMos D1

### DISCUSSION THREAD

**Discussion by pirracas77** [arendst/Tasmota#11963] [Ideas]
Hello,

I have this ultrasonic sensor.

https://es.aliexpress.com/item/4001287816907.html

I thought that is going to work the same as https://tasmota.github.io/docs/HC-SR04/ but I'm wrong.

 Do you have any advice to use it with tasmota?

**Comment by sfromis** [comment]
The product page indicates outputs via UART, PWM or "switch quantity". DFRobot has [more info](https://wiki.dfrobot.com/A02YYUW%20Waterproof%20Ultrasonic%20Sensor%20SKU:%20SEN0311) about the sensor, including serial protocol.

**Reply by pirracas77** [reply]
Thanks for the answer.
I have one UART. I'm very noob and have no idea how to get works in Tasmota with the information explained in DFRobot. 
Could you put me on the track?

**Reply by barbudor** [reply]
https://tasmota.github.io/docs/Commands/#serial-bridge

**Reply by pirracas77** [reply]
With this data

UART | data bit | stop bit | parity | band rate
TTL level | 8 | 1 | none | 9600bps

I think I have to set up:

Baudrate 9600
SerialConfig 8n1

20:18:29.251 CMD: Baudrate 9600
20:18:29.254 SER: Set to 8N1 9600 bit/s
20:18:29.259 RSL: RESULT = {"Baudrate":9600}


Right?

But nothing is showing in console. 
20:20:29.928 CMD: 
20:24:06.931 CMD:

In fact I can see stranges characteres at higher baudrate.

20:18:00.311 RSL: RESULT = {"Baudrate":115200}
20:18:01.503 CMD: 
20:18:02.844 CMD: 
20:18:03.121 CMD: 
20:18:07.028 CMD: @

**Reply by sfromis** [reply]
Do use `SerialLog 0` to turn off Tasmota seeing serial as a console. You may want to issue a `SerialSend5` command to get data in hex mode instead of chars. Otherwise, you're not likely to see anything useful.

**Reply by pirracas77** [reply]
Hello and thanks for you support.
This my today's configuration.
07:57:21.754 CMD: SBaudrate 1
07:57:21.759 RSL: RESULT = {"SBaudrate":9600}
07:57:46.736 CMD: SerialLog 0
07:57:46.740 RSL: RESULT = {"SerialLog":{"0":{"Active":"0"}}}
07:58:01.692 CMD: SerialSend 5
07:58:01.696 RSL: RESULT = {"SerialSend":"Done"}
08:01:40.153 CMD: SerialConfig 8n1
08:01:40.157 RSL: RESULT = {"SerialConfig":"8N1"}


![image](https://user-images.githubusercontent.com/18216098/116774239-8b417200-aa5b-11eb-9fc9-48387187b56f.png)


But nothing happens in the console.

I'm doing somthing wrong maybe. Any idea?

**Reply by OnkelTom2805** [reply]
Hello,

I am struggling with exact the same problem.

Did you solve it in the meantime ?

Any help is much appreciated !

Onkel Tom

**Reply by pirracas77** [reply]
yes, see the next post.

**Reply by OnkelTom2805** [reply]
Indeed. That works. 

(I had read it, but first thought this could not be the solution as the SR04 has a different interface from the A02YYUW, however, it works !)

Many thanks gain.

**Reply by PG430** [reply]
how to change measurement threshold? Sensor is not responding after transmitting threshold value

**Reply by alexandrechoske** [reply]
Can you give more details on how you get this working? IM struggling here

**Comment by pirracas77** [comment]
No idea what I did but it is working now with the SR04 configuration.

![image](https://user-images.githubusercontent.com/18216098/116774869-cc3b8580-aa5f-11eb-8707-71b81473445c.png)
![image](https://user-images.githubusercontent.com/18216098/116774880-e37a7300-aa5f-11eb-9ae4-468e433c3523.png)


Maybe a mix beetween SR04 selection and baudrate adjust? Mixing wires?

**Comment by pirracas77** [comment]
I put an schema for everybody that could find problems with this specific sensor.

![image](https://user-images.githubusercontent.com/18216098/116777815-b6818c80-aa6e-11eb-8e42-fe36a34c1e55.png)

**Reply by Scoff123** [reply]
I've just ordered one of these sensors, but the PWM version rather than the UART version. I'm hoping to connect it to either my ESP32 devkit board or wemos D1 mini, both of which require a 3.3v signal form the sensor. Is this what your resistors in the above picture are doing - reducing the voltage from 5v down to the necessary 3.3v for your sensor? From what I've read, the echo pin from the sensor requires a voltage level shifter before going into the ESP board. Is that how your resistors are connected? Sorry I can't quite make it out on the picture. Do you have any pictures of your finished wiring without the breadboard? Trying to work out the neatest way of doing this. Thanks!

**Reply by pirracas77** [reply]
You have to follow this schema
https://tasmota.github.io/docs/HC-SR04/

**Comment by Scoff123** [comment]
Thank you

**Comment by topsurfer** [comment]
Hi, what is now the bether choice to order:

The PCM or the UART version?

Will use it with Tasmota "as" a SR04....

**Reply by sfromis** [reply]
Not PWM, the HC-SR04 use UART mode....

**Reply by topsurfer** [reply]
Thx!

**Comment by wubbl0rz** [comment]
attached the same sensor this morning directly to the pins of an esp32 nodemcu. working flawlessly with the sr04 settings. if you have problems make sure to swap rx/tx. first i made the error and connected green tx cable to tx pin. after swapping it worked immediately.

**Comment by bauerpower01** [comment]
To help people find this in the future. It also works for the A01NYUB ultrasonic sensor. I did not need the resistors to make it work like the HC-SR04. I was also using a EPS32-POE-ISO to power the sensor over POE.
